docs/serving/architecture.md

```diff
@@ -0,0 +1,52 @@
+# Knative Serving Architecture
+
+Knative Serving consists of several components forming the backbone of the Serverless Platform.
+This page explains the high-level architecture of Knative Serving. Please also refer to [the Knative Serving Overview](./README.md) 
+and [the Request Flow](./request-flow.md) for additional information.
+
+## Diagram
+
+![Knative Serving Architecture](images/serving-architecture.png)
+
+## Components
+
+| Component      | Responsibilities                                                                                                                                                                                                                                                                                                                                                 |
+|----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Activator      | The activator is part of the **data-plane**. It is responsible to queue incoming requests (if a `Knative Service` is scaled-to-zero). It communicates with the autoscaler to bring scaled-to-zero Services back up and forward the queued requests. Additional details can be found [here](https://github.com/knative/serving/blob/main/docs/scaling/SYSTEM.md). |
+| Autoscaler     | The autoscaler is responsible to scale the `Knative Services` based on configuration, metrics and incoming requests.                                                                                                                                                                                                                                             |
+| Controller     | It is the controller's job to manage the state of `Knative Resources` within the cluster. It watches several objects, manages the live cycle of dependent resources, and updates the resource state.                                                                                                                                                             |
+| Domain-Mapping | The domain-mapping controller is responsible to manage `DomainMapping` resources. It is an extension of the above `Controller`.                                                                                                                                                                                                                                  |
```

Comment:
Might mention that this is an alpha component and might or might not be installed.